gladier_kanzus/tools/gather_data.py

```diff
@@ -1,97 +1,116 @@
+    Variables:
+    - data['data_dir'] is the path where the raw (cbf) data is stored
+    - data['proc_dir'] is the path to where dials will run and save results
+    - data['upload_dir'] is the path to the folder which will be published into the portal
+    - data['trigger_name'] is the filename of the file that triggered the flow.
+    - data['exp'] is the experiment name
+    - data['sample'] is the protein/sample name
+    - data['chip_name'] is the chip namethe filename of the file that triggered the flow.
+    - data['run_num'] is the beamline run associated with this sample.
+    """
     import re
     import os
     import json
     import shutil
-    from zipfile import ZipFile
-    trigger_name = data['trigger_name']
+    import glob
+
     data_dir = data['data_dir']
-    processing_dir = data['proc_dir']
+    proc_dir = data['proc_dir']
     upload_dir = data['upload_dir']
-    tar_input = data['tar_input']
-
-    sample_metadata = trigger_name.split('/')[-4:]
-    assert len(sample_metadata) == 4 and trigger_name.endswith('cbf'), (
-        'Invalid sample path, must run with end path resembling: "S9/nsp10nsp16/K/Kaleidoscope_15_22016.cbf"'
-        f'Got: {trigger_name}'
-    )
 
-    # Gather metadata from path names
-    run_name, protein, _, trigger_file = sample_metadata
-    match = re.match(r'(\w+)_(\d+)_\d+.cbf', trigger_file)
-    if not match:
-        raise ValueError(f'Invalid trigger file: {trigger_file}, must match "Kaleidoscope_15_22016.cbf"')
-    exp_name, exp_number = match.groups()
+    trigger_name = data['trigger_name']
 
-    # Get processing and image dirs
-    # run_dir = os.path.dirname(trigger_name)
+    ##experiment information
+    run_name = data['exp']
+    protein= data['sample']
+    chip_name = data['chip_name']
+    run_num = data['run_num']
 
-    # Check it exists in case xy search didn't create this one
-    if not os.path.exists(processing_dir):
-        os.mkdir(processing_dir)
+    beamline_file_name = f'beamline_run{run_num}.json'
+    beamline_file = os.path.join(data_dir, beamline_file_name)
 
-    beamline_file = os.path.join(data_dir, f'beamline_run{exp_number}.json')
     if not os.path.exists(upload_dir):
         os.mkdir(upload_dir)
-    shutil.copyfile(beamline_file, os.path.join(upload_dir, os.path.basename(beamline_file)))
+    
+    if not os.path.isfile(os.path.join(upload_dir, beamline_file_name)):
+        shutil.copy( beamline_file, upload_dir)
 
-    # Fetch the list of ints that 'hit'.
     int_indices = []
     int_filenames = []
     cbf_indices = []
-    for filename in os.listdir(processing_dir):
-        # match int-0-Kaleidoscope_15_05189.pickle
-        int_match = re.match(r'int-(\d+)-(\w+)_(\d+)_(\d+).pickle', filename)
+
+    for int_file in glob.glob(os.path.join(proc_dir,'int-*.pickle')):
+        int_file = os.path.basename(int_file)
+        int_match = re.match(r'int-\d+-\w+_\d+_(\d+).pickle', int_file)
```

Comment:
I think this regex is also used in the plotting step. A likely scenario is this will change during a beamline for some reason. It may be good to pass this as input and re-use it anywhere else its needed. Ideally, we'd only gather the int files once.